.github/workflows/pullrequests.yml

```diff
@@ -12,41 +12,32 @@ jobs:
       fail-fast: false
       matrix:
         python:
-        - 3.7
         - 3.8
         - 3.9
         - "3.10"
         - 3.11
         libraries:
         - marshmallow=='3.0.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.1.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.2.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.3.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.4.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.5.*' flask=='2.0.*' werkzeug=='2.0.*'
         - marshmallow=='3.10.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.11.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.12.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.13.*' flask=='2.0.*' werkzeug=='2.0.*'
-        - marshmallow=='3.14.*' flask=='2.0.*' werkzeug=='2.0.*'
         - marshmallow=='3.15.*' flask=='2.0.*' werkzeug=='2.0.*'
+        - marshmallow=='3.19.*' flask=='2.0.*' werkzeug=='2.0.*'
         - marshmallow=='3.0.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.1.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.2.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.3.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.4.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.5.*' flask=='2.1.*' werkzeug=='2.1.*'
         - marshmallow=='3.10.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.11.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.12.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.13.*' flask=='2.1.*' werkzeug=='2.1.*'
-        - marshmallow=='3.14.*' flask=='2.1.*' werkzeug=='2.1.*'
         - marshmallow=='3.15.*' flask=='2.1.*' werkzeug=='2.1.*'
+        - marshmallow=='3.19.*' flask=='2.1.*' werkzeug=='2.1.*'
+        - marshmallow=='3.0.*' flask=='2.2.*' werkzeug=='2.2.*'
+        - marshmallow=='3.10.*' flask=='2.2.*' werkzeug=='2.2.*'
+        - marshmallow=='3.15.*' flask=='2.2.*' werkzeug=='2.2.*'
+        - marshmallow=='3.19.*' flask=='2.2.*' werkzeug=='2.2.*'
+        - marshmallow=='3.0.*' flask=='2.3.*' werkzeug=='2.3.*'
+        - marshmallow=='3.10.*' flask=='2.3.*' werkzeug=='2.3.*'
+        - marshmallow=='3.15.*' flask=='2.3.*' werkzeug=='2.3.*'
+        - marshmallow=='3.19.*' flask=='2.3.*' werkzeug=='2.3.*'
```

Comment:
Looks like a lot of combinations got dropped from the test matrix. Was this change all due to dropping 3.7 support?

Reply:
I dropped a lot because we have a limit on how many we can run in any given time, and i just don't know that arbitrarily testing all of them is useful. Especially when we are adding new flask and werkzeug versions.

Reply:
I just picked round numbers for marshmallow. and then 3.19.x because it has the new enum field.

Reply:
There is a maximum amount of matrix runs to 256.
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy

Reply:
Thinking about it; we should probably separate the Marshmallow versions from the Flask/Werkzuerg versions.

Looking back it seems like the following are the only versions where we should be checking against regressions
- Werzuerg 2.1 (dropped a method - https://github.com/plangrid/flask-rebar/pull/268)
- Marshmallow 3.18 (Introduces Enum)

Can we just re-write our matrix as
```
marshmallow:
 - 3.17.*
 - 3.18.*
 - 3.19.*
 flask:
 - flask=='2.0.*' werkzeug=='2.0.*'
 - flask=='2.1.*' werkzeug=='2.1.*'
 - flask=='2.3.*' werkzeug=='2.3.*'
```

Reply:
looks like it worked.